src/test/java/com/mdabrow9/ticketbookingapp/services/ReservationServiceImplTest.java

```diff
@@ -0,0 +1,256 @@
+            seat.setId((long) i);
+            seat.setSeatRow(0);
+            seat.setSeatNumber(i);
+            seat.setRoom(room);
+            seats.add(seat);
+        }
+
+        screening = new Screening();
+        screening.setId(1L);
+        screening.setMovie(movie);
+        movie.setScreenings(List.of(screening));
+
+        screening.setStartTime(LocalDateTime.of(2021,7,20,12,0));
+        screening.setRoom(room);
+        room.setScreenings(List.of(screening));
+
+        ticketType = new TicketType("Adult",new BigDecimal("25.0"));
+
+        seatReservations = new ArrayList<>();
+        /*SeatReservation  sR = new SeatReservation();
+        sR.setId(1L);
+        sR.setScreening(screening);
+        sR.setSeat(seats.get(3));*/
+
+        Mockito.when(screeningRepository.findAllById(anyLong())).thenReturn(Optional.of(screening));
+
+        for( Seat s:seats)
+        {
+            Mockito.when(seatRepository.findById(eq(s.getId()))).thenReturn(Optional.of(s));
+        }
+        Mockito.when(seatReservationRepository.findByScreeningAndSeat(any(), any())).thenReturn(Optional.empty());
+
+        Mockito.when(seatReservationRepository.findAllByRowAndScreening(anyInt(), any())).thenReturn(seatReservations);
+        Mockito.when(ticketTypeRepository.findByName(anyString())).thenReturn(Optional.of(ticketType));
+
+
+    }
+
+    @Test
+    public void createReservationTest()
+    {
+
+
+        ReservationResponseDTO response = reservationService.createReservation(request.getSeatId(),request.getScreeningId(),request.getName(),request.getSurname(),request.getReservationTickets());
+
+        assertNotEquals(null,response);
+        assertEquals(BigDecimal.valueOf(75.0),response.getTotalCost());
+        assertEquals(screening.getStartTime().minusMinutes(10),response.getExpirationTime());
+    }
+
+    @Test(expected = ConflictException.class)
+    public void SingleSlotLeftBetweenSeatsTest()
+    {
+
+        request.setSeatId(new Long[]{1L,2L,4L});
+
+        ReservationResponseDTO response = reservationService.createReservation(request.getSeatId(),request.getScreeningId(),request.getName(),request.getSurname(),request.getReservationTickets());
+
+
+    }
+
+    @Test(expected = ConflictException.class)
+    public void fifteenMinutesBeforeScreeningTest()
+    {
+        screening.setStartTime(LocalDateTime.now());
+
+        ReservationResponseDTO response = reservationService.createReservation(request.getSeatId(),request.getScreeningId(),request.getName(),request.getSurname(),request.getReservationTickets());
+
+
+    }
+
+    @Test(expected = ConflictException.class)
+    public void wrongClientName1()
+    {
+        request.setName("jan");
+
+        ReservationResponseDTO response = reservationService.createReservation(request.getSeatId(),request.getScreeningId(),request.getName(),request.getSurname(),request.getReservationTickets());
+    }
+    @Test(expected = ConflictException.class)
```

Comment:
to nie jest najlepszy sposób łapania wyjątków w testach - nowy junit ma lepszą metodę